src/replica/duplication/duplication_sync_timer.cpp

```diff
@@ -113,12 +117,28 @@ void duplication_sync_timer::update_duplication_map(
 {
     for (replica_ptr &r : get_all_replicas()) {
         auto it = dup_map.find(r->get_gpid().get_app_id());
+        app_info *info = const_cast<app_info *>(r->get_app_info());
+        gpid id = r->get_gpid();
+        bool doing_duplication;
+        if (replica_is_cloing_or_closed(id)) {
+            continue;
+        }
+
         if (it == dup_map.end()) {
             // no duplication is assigned to this app
             r->get_duplication_manager()->update_duplication_map({});
+            doing_duplication = false;
         } else {
             r->get_duplication_manager()->update_duplication_map(it->second);
+            doing_duplication = true;
+        }
+
+        if (doing_duplication) {
+            info->__set_duplicating(true);
```

Comment:
It would be better to update the info in replica class to avoid concurrent write, see replica::update_app_max_replica_count() for example.